include/drivers/stepper.h

```diff
@@ -0,0 +1,140 @@
+/**
+ * @file
+ *
+ * @brief Public APIs for the stepper controller drivers.
+ */
+
+/*
+ * Copyright (c) 2020 Teslabs Engineering S.L.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_STEPPER_H_
+#define ZEPHYR_DRIVERS_STEPPER_H_
+
+#include <zephyr/types.h>
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @cond INTERNAL_HIDDEN
+ *
+ * These are for internal use only, so skip these in
+ * public documentation.
+ */
+
+struct stepper_driver_api {
+	int (*set_enabled)(struct device *dev, uint32_t channel, bool enabled);
+	int (*set_speed)(struct device *dev, uint32_t channel, int32_t speed);
+	int (*move)(struct device *dev, uint32_t channel, int32_t steps);
+#ifdef CONFIG_STEPPER_ASYNC
+	int (*move_async)(struct device *dev, uint32_t channel, int32_t steps,
+			  struct k_poll_signal *async);
+#endif
+	int (*stop)(struct device *dev, uint32_t channel);
+};
+
+/**
+ * @endcond
+ */
+
+/**
+ * @brief Enable or disable a stepper channel.
+ * @param dev	  Stepper controller device.
+ * @param channel Stepper channel.
+ * @param enabled Enabled flag.
+ */
+__syscall int stepper_set_enabled(struct device *dev, uint32_t channel,
+				  bool enabled);
+
+static inline int z_impl_stepper_set_enabled(struct device *dev,
+					     uint32_t channel, bool enabled)
+{
+	const struct stepper_driver_api *api =
+		(const struct stepper_driver_api *)dev->driver_api;
+	return api->set_enabled(dev, channel, enabled);
+}
+
+/**
+ * @brief Configure stepper speed.
+ *
+ * @param dev	  Stepper controller device.
+ * @param channel Stepper channel.
+ * @param speed   Speed (steps/s).
```

Comment:
If this can fail (e.g. the speed is not supported) the API should return `int` and document potential error codes.

What is the semantics of a negative speed? Direction?

What is the behavior on a zero speed?